src/coreclr/jit/hwintrinsiclistarm64sve.h

```diff
@@ -95,6 +95,18 @@ HARDWARE_INTRINSIC(Sve,           LoadVectorUInt16ZeroExtendToUInt32,
 HARDWARE_INTRINSIC(Sve,           LoadVectorUInt16ZeroExtendToUInt64,                               -1,      2,      false, {INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_sve_ld1h,       INS_invalid,        INS_invalid},     HW_Category_MemoryLoad,            HW_Flag_Scalable|HW_Flag_ExplicitMaskedOperation|HW_Flag_LowMaskedOperation)
 HARDWARE_INTRINSIC(Sve,           LoadVectorUInt32ZeroExtendToInt64,                                -1,      2,      false, {INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_sve_ld1w,       INS_invalid,        INS_invalid,        INS_invalid},     HW_Category_MemoryLoad,            HW_Flag_Scalable|HW_Flag_ExplicitMaskedOperation|HW_Flag_LowMaskedOperation)
 HARDWARE_INTRINSIC(Sve,           LoadVectorUInt32ZeroExtendToUInt64,                               -1,      2,      false, {INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_sve_ld1w,       INS_invalid,        INS_invalid},     HW_Category_MemoryLoad,            HW_Flag_Scalable|HW_Flag_ExplicitMaskedOperation|HW_Flag_LowMaskedOperation)
+HARDWARE_INTRINSIC(Sve,           LoadVectorByteNonFaultingZeroExtendToInt16,                       -1,      1,      false, {INS_invalid,        INS_invalid,        INS_sve_ldnf1b,     INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid},     HW_Category_MemoryLoad,            HW_Flag_Scalable|HW_Flag_EmbeddedMaskedOperation|HW_Flag_LowMaskedOperation|HW_Flag_SpecialCodeGen)
```

Comment:
Maybe I'm missing something....

`HW_Flag_SpecialCodeGen` is set for these.
During codegen, `genHWIntrinsic()` is called with `LoadVectorByteNonFaultingZeroExtendToInt16`.
`intrin.codeGenIsTableDriven()` check fails (due to `HW_Flag_SpecialCodeGen`).
Code falls into the `switch (intrin.id)` at the end of the function.
Switch hits `default: unreached()` due to no `LoadVectorByteNonFaultingZeroExtendToInt16` case

Reply:
Because of the "HW_Flag_EmbeddedMaskedOperation" flag, at the point of `genHWIntrinsic`, `intrin` will never be `Sve_LoadVectorByteNonFaultingZeroExtendToInt16` and instead be `Sve_ConditionalSelect` that wraps `Sve_LoadVectorByteNonFaultingZeroExtendToInt16`. This is why I had to handle `intrinEmbMask.codeGenIsTableDriven()` like you saw.